larpandoracontent/LArHelpers/LArGeometryHelper.cc

```diff
@@ -394,11 +394,26 @@ CartesianVector LArGeometryHelper::GetWireAxis(const Pandora &pandora, const Hit
 
 bool LArGeometryHelper::IsInGap(const Pandora &pandora, const CartesianVector &testPoint2D, const HitType hitType, const float gapTolerance)
 {
-    // ATTN: input test point MUST be a 2D position vector 
+    // ATTN: input test point MUST be a 2D position vector
     for (const DetectorGap *const pDetectorGap : pandora.GetGeometry()->GetDetectorGapList())
     {
-        if (pDetectorGap->IsInGap(testPoint2D, hitType, gapTolerance))
-            return true;
+        // Check line gaps (need to input 2D hit type)
+        const LineGap *const pLineGap = dynamic_cast<const LineGap*>(pDetectorGap);
+
+        if (pLineGap)
+        {
+            if (pLineGap->IsInGap(testPoint2D, hitType, gapTolerance))
+                return true;
+        }
+
+        // Check box gaps (need to input 3D hit type)
+        const BoxGap *const pBoxGap = dynamic_cast<const BoxGap*>(pDetectorGap);
+
+        if (pBoxGap)
+        {
+            if (pBoxGap->IsInGap(testPoint2D, TPC_3D, gapTolerance))
+                return true;
+        }
```

Comment:
I don't think we can proceed with this as a valid solution - it is working hard to subvert the original, intended use of detector gaps, with a IsInGap function in a purely abstract base class.

I think it would all be much simpler if we modify the LineGap class in the SDK to hold information about whether the gap is in the drift direction or the wire direction.

All of these (slow, often deprecated) dynamic casts should disappear as would the separate treatments for LineGaps and BoxGaps below.

BoxGaps could indeed be used for work that is truly in a 3D coordinate system.

The LineGap modification would not take long at all, and there would need to be an associated tweak to the display of LineGaps in PandoraMonitoring.

What do you think?